lib/src/view/user/game_history_screen.dart

```diff
@@ -183,3 +259,203 @@ class _BodyState extends ConsumerState<_Body> {
     );
   }
 }
+
+class _FilterGames extends ConsumerStatefulWidget {
+  const _FilterGames({
+    required this.filter,
+    required this.user,
+  });
+
+  final GameFilterState filter;
+  final LightUser? user;
+
+  @override
+  ConsumerState<_FilterGames> createState() => _FilterGamesState();
+}
+
+class _FilterGamesState extends ConsumerState<_FilterGames> {
+  late GameFilterState filter;
+
+  @override
+  void initState() {
+    super.initState();
+    filter = widget.filter;
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    const gamePerfs = [
+      Perf.ultraBullet,
+      Perf.bullet,
+      Perf.blitz,
+      Perf.rapid,
+      Perf.classical,
+      Perf.correspondence,
+      Perf.chess960,
+      Perf.antichess,
+      Perf.kingOfTheHill,
+      Perf.threeCheck,
+      Perf.atomic,
+      Perf.horde,
+      Perf.racingKings,
+      Perf.crazyhouse,
+    ];
+    const filterGroupSpace = SizedBox(height: 10.0);
+
+    final session = ref.read(authSessionProvider);
+    final userId = widget.user?.id ?? session?.user.id;
+
+    List<Perf> availablePerfs(User user) {
+      final perfs = gamePerfs.where((perf) {
+        final p = user.perfs[perf];
+        return p != null && p.numberOfGamesOrRuns > 0;
+      }).toList(growable: false);
+      perfs.sort(
+        (p1, p2) => user.perfs[p2]!.numberOfGamesOrRuns
+            .compareTo(user.perfs[p1]!.numberOfGamesOrRuns),
+      );
+      return perfs;
+    }
+
+    Widget perfFilter(List<Perf> choices) => _Filter<Perf>(
+          filterName: context.l10n.variant,
+          icon: const Icon(LichessIcons.classical),
+          filterType: FilterType.multipleChoice,
+          choices: choices,
+          choiceSelected: (choice) => filter.perfs.contains(choice),
+          choiceLabel: (t) => t.title,
+          onSelected: (value, selected) => setState(
+            () {
+              filter = filter.copyWith(
+                perfs: selected
+                    ? filter.perfs.add(value)
+                    : filter.perfs.remove(value),
+              );
+            },
+          ),
+        );
+
+    return Container(
+      padding: const EdgeInsets.all(16),
+      child: DraggableScrollableSheet(
+        initialChildSize: .7,
+        expand: false,
+        snap: true,
+        snapSizes: const [.7],
+        builder: (context, scrollController) => ListView(
+          controller: scrollController,
+          children: [
+            if (userId != null)
+              ref.watch(userProvider(id: userId)).when(
```

Comment:
It is better to put `ref.watch` at the beginning of `build()` methods so when we read the code we know immediately what are the watched dependencies of the widget.

Reply:
The `userId` might be null though, so we can't listen to it unconditionally.